File: POO/aula18/Divisao/Divisao.cpp
#include <iostream>
#include <stdexcept>

class DivideByZeroException : public std::runtime_error
{
	public:
		DivideByZeroException() : runtime_error("Erro: Divisao por zero!")
		{	}
};

float divisao(int a, int b) throw(DivideByZeroException)
{
	if(b != 0)
	{
		return a/(float)b;
	}
	else
	{
		throw DivideByZeroException();
	}
}

int main()
{
	int a, b;
	while(std::cin >> a >> b)
	{
		try
		{
			std::cout << "Resultado: " << divisao(a, b) << std::endl;
		}
		catch(DivideByZeroException dbze)
		{
			std::cerr << dbze.what() << std::endl;
		}
	}
	return 0;
}

File: POO/aula18/JDivisao/Divisao.java
import java.util.Scanner;

public class Divisao
{
	public static void main(String[] args)
	{
		int a, b;
		Scanner scan = new Scanner(System.in);
		while(scan.hasNext())
		{
			a = scan.nextInt();
			b = scan.nextInt();
			try
			{
				System.out.println("Resultado: " + divisao(a, b));
			}
			catch(DivideByZeroException dbze)
			{
				System.err.println(dbze.getMessage());
			}
		}
	}

	public static float divisao(int a, int b) throws DivideByZeroException
	{
		if(b != 0)
		{
			return a/(float)b;
		}
		else
		{
			throw new DivideByZeroException();
		}
	}
}

class DivideByZeroException extends Exception
{
	public DivideByZeroException()
	{
		super("Erro: Divisao por zero!");
	}
}
